test/e2e/framework/service/jig.go

```diff
@@ -901,10 +901,17 @@ func (j *TestJig) checkNodePortServiceReachability(svc *v1.Service, pod *v1.Pod)
 // checkExternalServiceReachability ensures service of type externalName resolves to IP address and no fake externalName is set
 // FQDN of kubernetes is used as externalName(for air tight platforms).
 func (j *TestJig) checkExternalServiceReachability(svc *v1.Service, pod *v1.Pod) error {
+	// NOTE(claudiub): Windows does not support PQDN.
+	svcName := svc.Name
+	if framework.NodeOSDistroIs("windows") {
+		svcName = fmt.Sprintf("%s.%s.svc.%s", svcName, svc.Namespace, framework.TestContext.ClusterDNSDomain)
+	}
 	// Service must resolve to IP
-	cmd := fmt.Sprintf("nslookup %s", svc.Name)
-	_, err := framework.RunHostCmd(pod.Namespace, pod.Name, cmd)
-	if err != nil {
+	cmd := fmt.Sprintf("nslookup %s", svcName)
+	_, stderr, err := framework.RunHostCmdWithFullOutput(pod.Namespace, pod.Name, cmd)
+	// NOTE(claudiub): nslookup may return 0 on Windows, even though the DNS name was not found. In this case,
+	// we can check stderr for the error.
+	if err != nil || strings.Contains(stderr, fmt.Sprintf("can't find %s", svcName)) {
```

Comment:
This seems potentially error prone / flaky,  I'm curious why nslookup behavior is like this on Windows and if there's a more consistent way to check failure.

Reply:
Windows and Linux commands behave quite differently, that is an already known fact. In this case, the Windows' ``nslookup`` doesn't have a return code, which means it will always be "succesful".

We've replaced it with bind-tools' ``nslookup`` which properly returns a non-zero value if the given DNS name is not found.

With this change here, we're trying to make sure that we properly fail the test for the Windows' ``nslookup`` by checking its stderr. Here's the output we're checking against:

    Apr 28 21:25:06.371: INFO: Running '/home/ubuntu/run_e2e/kubernetes/_output/dockerized/bin/linux/amd64/kubectl --server=https://kubetest-63c87972-83ef-11ea-ae60-024273197af3.westus2.cloudapp.azure.com --kubeconfig=/home/ubuntu/kube/config_200424 exec --namespace=services-2633 agnhost-pod -- /bin/sh -x -c nslookup nodeport-service'
    Apr 28 21:25:09.610: INFO: stderr: "+ nslookup nodeport-service\n*** kube-dns.kube-system.svc.cluster.local can't find nodeport-service: Non-existent domain\r\r\n"
    Apr 28 21:25:09.610: INFO: stdout: "Server:  kube-dns.kube-system.svc.cluster.local\r\nAddress:  10.0.0.10\r\n\r\n"